makefiles/utils.inc.mk

```diff
@@ -0,0 +1,87 @@
+# Assorted make utilities
+# TODO: move tests out of here
+
+test-all-utils: test-streq test-assert-positive test-assert_eq-positive
+
+# Return an empty string if the arguments are different (works only for single
+# words
+streq = $(filter $(1),$(2))
+
+test-streq:
+	@echo empty,empty: $(call streq,,)
+	@echo empty,x: $(call streq,,x)
+	@echo x, empty: $(call streq,x,)
+	@echo x, y: $(call streq,abc,abcw)
+	@echo y, x: $(call streq,abcw,abc)
+	@echo x,x: $(call streq,abcw,abcw)
+
+# Produce an error if the argument is empty. The second argument defines the
+# message.
+assert = $(if $(1),,$(error $(2)))
+
+test-assert-positive:
+	@echo OK? $(call assert,x,"This should not fail")
+
+test-assert-negative:
+	@echo OK? $(call assert,,"This should fail")
+
+# Produce an error if the first two arguments are different. The third argument
+# defines the message. If the message is empty a default message is shown.
+assert_eq = $(if $(call streq,$(1),$(2)),,$(error $(if $(3),$(3),Expected $(2) and got $(1))))
+
+test-assert_eq-positive:
+	@echo OK? $(call assert_eq,x,x,"This should not fail")
+
+test-assert_eq-negative:
+	@echo OK? $(call assert_eq,z,x,"This should fail")
+
+# Get the remaining words in a list
+rest = $(wordlist 2,$(words $(1)),$(1))
```

Comment:
Some functions like this one would maybe better be private namespaced functions as the name is not really explicit and could collide with another name.